File: api/lib/scribe/line.ex
defmodule Scribe.Line do
  use Ecto.Schema

  schema "lines" do
    field :text, :string
    field :email, :string
    field :name, :string
  end

  def changeset(line, params \\ %{}) do
    line
      |> Ecto.Changeset.cast(params, [:text, :email, :name])
      |> Ecto.Changeset.validate_required([:text, :email])
  end
end

File: api/src/domain/story/line/line.rs
// -- types --
#[derive(Debug)]
pub struct Line {
    pub text: String,
    pub name: Option<String>,
    pub email: Option<String>,
}

// -- impls --
impl Line {
    pub fn new(text: String, name: Option<String>, email: Option<String>) -> Self {
        Line {
            text: text,
            name: name,
            email: email,
        }
    }
}
